fetch/mesh_data.go

```diff
@@ -201,68 +201,63 @@ func (f *Fetch) GetPoetProof(ctx context.Context, id types.Hash32) error {
 	}
 }
 
-func (f *Fetch) GetMaliciousIDs(
-	ctx context.Context,
-	peers []p2p.Peer,
-	okCB func([]byte, p2p.Peer),
-	errCB func(error, p2p.Peer),
-) error {
-	return poll(ctx, f.servers[malProtocol], peers, []byte{}, okCB, errCB)
+func (f *Fetch) GetMaliciousIDs(ctx context.Context, peers []p2p.Peer) <-chan Result {
+	return poll(ctx, f.servers[malProtocol], peers, []byte{})
 }
 
 // GetLayerData get layer data from peers.
 func (f *Fetch) GetLayerData(
 	ctx context.Context,
 	peers []p2p.Peer,
 	lid types.LayerID,
-	okCB func([]byte, p2p.Peer),
-	errCB func(error, p2p.Peer),
-) error {
+) (<-chan Result, error) {
 	lidBytes, err := codec.Encode(&lid)
 	if err != nil {
-		return err
+		return nil, err
 	}
```

Comment:
I believe a `codec.MustEncode()` could be used here. It would simplify the interface (no error returned).

Reply:
Never mind, this is for the request not the response, so `MustEncode` is probably fine.